http-core/src/main/scala/org/apache/pekko/http/impl/engine/parsing/HttpMessageParser.scala

```diff
@@ -307,17 +307,20 @@ private[http] trait HttpMessageParser[Output >: MessageOutput <: ParserOutput] {
         s"HTTP chunk extension length exceeds configured limit of ${settings.maxChunkExtLength} characters")
 
     @tailrec def parseSize(cursor: Int, size: Long): StateResult =
-      if (size <= settings.maxChunkSize) {
+      if (size <= Int.MaxValue) {
```

Comment:
My personal preference would be to have:
```
if (size > Int.MaxValue) {
// early exit here with failEntityStream
}

// everything else

```

But I don't feel is binding.

Reply:
guard clauses. 👍🏻

Reply:
I'm trying to keep the structure consistent with the rest of the file. I also think the above suggestion would be incorrect in the sense that `everything else` would still be executed after calling `failEntityStream`? An else block would still be required.